SystemManager/Src/SystemManager.cpp

```diff
@@ -10,37 +10,56 @@
 #include "drivers_config.hpp"
 #include "independent_watchdog.h"
 #include "tim.h"
+#include "CommonDataTypes.hpp"
+#include "config.hpp"
+#include "cmsis_os.h"
 
 SystemManager::SystemManager():
     rcController_(&huart2),
     throttleMotorChannel_(&htim2, TIM_CHANNEL_1),
     yawMotorChannel_(&htim2, TIM_CHANNEL_2),
     rollMotorChannel_(&htim2, TIM_CHANNEL_3),
     pitchMotorChannel_(&htim2, TIM_CHANNEL_4),
-    invertedRollMotorChannel_(&htim3, TIM_CHANNEL_1),
-    watchdog_(&hiwdg)
-{}
+    watchdog_(&hiwdg),
+    am_instance_(config::flightmodes[config::DEFAULT_FLIGHTMODE].flightmodeConstructor())
+{
+    xTaskCreate(runAM, "AM Thread", 400U, (void*)&am_instance_, osPriorityNormal, &AM_handle_);
+}
+
+SystemManager::~SystemManager() {
+    vTaskDelete(AM_handle_);
+}
 
-SystemManager::~SystemManager() {}
+void SystemManager::runAM(void* pvParameters) {
+    AM::AttitudeManager* am_instance = (AM::AttitudeManager *) pvParameters;
+    TickType_t xLastWakeTime;
+    xLastWakeTime = xTaskGetTickCount();
+    for(;;) {
+        am_instance->runControlLoopIteration(AM::AttitudeManager::getControlInputs());
+        vTaskDelayUntil(&xLastWakeTime, AM_PERIOD_MS);
+    }
+}
 
 void SystemManager::flyManually() {
     for(;;){
```

Comment:
Where would this be called? I imagine it would ideally run repeatedly at the same rate as runAM so that it has up-to-date control inputs.

Reply:
This is called in `main.cpp` when we create a SM thread that calls `flyManually()`